### DISCUSSION TITLE
Integrate TRMM with existing Meshcentral server

### DISCUSSION THREAD

**Discussion by VINISHVV** [amidaware/tacticalrmm#1133] [General]
Currently we have deployed a meshcentral server with 9000+ customized meshagent. We need to integrate the same with TRRM server.can you suggest a solution to integrate the same

**Comment by dinger1986** [comment]
Would you like to keep your existing mesh server? https://docs.tacticalrmm.com/mesh_integration/#running-your-own-existing-or-separate-meshcentral-server

Or something else?

**Reply by VINISHVV** [reply]
Exisging server with cusgomized agent

**Reply by dinger1986** [reply]
No reason you can't use your existing custom mesh agents as shared previously the doc link.

I'm confused what the question is, please give a clear and concise description/question.

**Comment by VINISHVV** [comment]
Hi,
My agent is customized means its agent installed at different location.

Tactical rmm looks the agent availability at c:/programfiles/meshagent folder by default.

In my case the installation location is different

**Comment by wh1te909** [comment]
The installer supports the `-meshdir` optional argument where you can specify the path to your mesh directory: https://docs.tacticalrmm.com/install_agent/#optional-installer-args

**Reply by Mfinadmin** [reply]
@wh1te909 - I just gone through the Link. It seems the installation location is works fine, But in my case the mesh agent name is also different

**Reply by dinger1986** [reply]
ok, does that mean it doesnt work?

**Reply by Mfinadmin** [reply]
Is it possible to add the agent name since reinstallation of 9K device is little difficult.

**Reply by dinger1986** [reply]
I presume you have tried it and it doesnt work? if thats the case then I would suggest opening a feature request to implement that

**Reply by Mfinadmin** [reply]
The installation folder works fine, But the Meshagent.exe is searching by TRMM.

Due to above issues its not worked form me.

**Reply by dinger1986** [reply]
ok, are you happy opening a feature request then?

**Reply by Mfinadmin** [reply]
Sure, Will raise a new task
